### PR TITLE
Added Functionality : Sword and Basilisk(Monster)

### PR DESCRIPTION
**Pull request by saarthakchats**
### Changes done in this Pull Request

- 
  Related Issues or Pull Requests #<Issue no.>

### CODE REVIEW

**Comment by saarthakchats** [comment]
@niccokunzmann Pls merge :D

File: js/tiles.js [hunk 1]
```diff
@@ -974,11 +983,39 @@ const door = {
...
         createImages: function() {
           this.wallTop = this.createImage("tiles/rooms/door/top.svg");
           this.wallRight = this.createImage("tiles/rooms/wall/right.svg");
           this.ground = this.createImage("tiles/rooms/floor/HandDrawnTile.svg");
-           alertInfo("This tile was Hand-Drawn by Beta-King", "");
+        },
+        visit: function() {
+             if (player.inventory.has('Sword')) {
+             alertSuccess("You killed the Monster! Yay!!");
+             player.inventory.remove('Spoon');
```
Comment (niccokunzmann) [inline js/tiles.js]: why do you remove the spoon?

Comment (niccokunzmann) [reply js/tiles.js]: How about is drops some item, just an idea, nothing blocking.

File: js/tiles.js [hunk 2]
```diff
@@ -974,11 +983,39 @@ const door = {
         }
     }),
     threeHeads: Object.assign({}, OpenDoors, {
+    	canEnterFromTheRight() {
```
Comment (niccokunzmann) [inline js/tiles.js]: You can change all these functions but please also change the look. Currently, it can be entered from left, top, bottom.

**Comment by saarthakchats** [comment]
@niccokunzmann Done changes

**Comment by Beta-King** [comment]
please give a preview link :D

**Comment by saarthakchats** [comment]
@Beta-King 
The game is currently down

**Comment by Beta-King** [comment]
Oh is that so ?? kk nevermind then, ill check it in code basis :D

**Comment by saarthakchats** [comment]
Yes please :D

**Comment by Beta-King** [comment]
It doesnt look good, could you change that ??

File: js/tiles.js [hunk 3]
```diff
@@ -974,11 +983,39 @@ const door = {
...
           this.wallRight = this.createImage("tiles/rooms/wall/right.svg");
           this.ground = this.createImage("tiles/rooms/floor/HandDrawnTile.svg");
-           alertInfo("This tile was Hand-Drawn by Beta-King", "");
+        },
+        visit: function() {
+             if (player.inventory.has('Sword')) {
+             alertSuccess("You killed the Monster! Yay!!");
+             player.inventory.remove('Sword');
+            } else {
+                alertNormal("You Need a Sword to fight the Basilisk!", "It is deadly");
```
Comment (Beta-King) [inline js/tiles.js]: this statement has no need as the player cant enter the tile anyways, all entries are locked until you get a sword right ?? so i suggest you open the entrance and design a badge and the player will only get the badge if he kills the monster, don't you think that'll make more sense ??

**Comment by saarthakchats** [comment]
@niccokunzmann 
I have made the changes you asked for

File: js/tiles.js (hunk 2 above)
Comment (niccokunzmann) [reply js/tiles.js]: @saarthakchats Please respond to this. I just want your view on this.

**Comment by niccokunzmann** [comment]
The game should work. At least the master branch works for me.

**Comment by saarthakchats** [comment]
Ya, can we merge

**Comment by saarthakchats** [comment]
@niccokunzmann

**Comment by niccokunzmann** [comment]
@saarthakchats I would like to ask you again, in case you did overlook this: 
There are a lot of forbidden entering directions. These should match the design.

- Is it really your intention to [show a door](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR1005) when you can not enter or [leave to the top](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR998) ?
- You do not show a wall on the left, yet [you forbid entering and leaving there](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR992) - is this really your intention?

If you can both answer with yes, I will merge and we create an issue to discuss this there. People who disagree might want to revert if they do not like to fix the code. 
Please answer these two questions, so I am sure this is considered.

**Comment by niccokunzmann** [comment]
There is one blocker: Please show that we have the rights to use the basilisk image.

**Comment by saarthakchats** [comment]
Taken from pixabay, added to CCO license folder in Images from Pixabay

**Comment by saarthakchats** [comment]
@niccokunzmann

**Comment by niccokunzmann** [comment]
Okay once again: Please answer to these two questions with yes and I will merge:


- Is it really your intention to [show a door](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR1005) when you can not enter or [leave to the top](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR998) ?
- You do not show a wall on the left, yet [you forbid entering and leaving there](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR992) - is this really your intention?

See https://github.com/fossasia/labyrinth/pull/495#issuecomment-357505949

**Comment by saarthakchats** [comment]
Yes

**Comment by saarthakchats** [comment]
We can create an issue after this if you want

**Comment by saarthakchats** [comment]
@niccokunzmann

**Comment by niccokunzmann** [comment]
So, you license the image under AGPL and under CC0. Okay. Please move the tile to that folder and license the image in the tile under CC0 and then remove the png file.

**Comment by niccokunzmann** [comment]
@saarthakchats Please create the issue.

**Comment by saarthakchats** [comment]
Can you give details of the issue

**Comment by saarthakchats** [comment]
@niccokunzmann

**Comment by niccokunzmann** [comment]
I need time to digest this conversation. Please ask in the chat what other people think.

**Comment by saarthakchats** [comment]
@niccokunzmann 
I have removed the basilisk image and added a snake drawn on inkscape, so the licensing issue doesnt exist anymore, Can we merge now?

**Comment by saarthakchats** [comment]
Is it okay now?
@niccokunzmann

File: js/tiles.js [hunk 4]
```diff
@@ -974,11 +983,38 @@ const door = {
         }
     }),
     threeHeads: Object.assign({}, OpenDoors, {
+    	canEnterFromTheRight() {
+            return false;
+        },
+        canLeaveToTheRight() {
+            return false;
+        },
+        canEnterFromTheLeft() {
```
Comment (niccokunzmann) [inline js/tiles.js]: please remove lines 992 to 996 to allow entering an leaving on the left.

File: js/tiles.js [hunk 5]
```diff
@@ -974,11 +983,38 @@ const door = {
...
+         },
+        canLeaveToTheLeft() {
+            return false;
+        },
+        canEnterFromTheTop: function(player) {
```
Comment (niccokunzmann) [inline js/tiles.js]: please remove lines 997 to 1002 to allow entering and leaving at the top.

File: js/tiles.js [hunk 6]
```diff
@@ -974,11 +983,38 @@ const door = {
...
+        },
         createImages: function() {
           this.wallTop = this.createImage("tiles/rooms/door/top.svg");
           this.wallRight = this.createImage("tiles/rooms/wall/right.svg");
           this.ground = this.createImage("tiles/rooms/floor/HandDrawnTile.svg");
-           alertInfo("This tile was Hand-Drawn by Beta-King", "");
+        },
+        visit: function() {
+             if (player.inventory.has('Sword')) {
+             alertSuccess("You killed the Monster! Yay!!");
```
Comment (niccokunzmann) [inline js/tiles.js]: Maybe you can say here that the sword got stuck ^^

**Comment by saarthakchats** [comment]
@niccokunzmann 
Sorry I left that, done now

File: js/tiles.js (hunk 6 above)
Comment (saarthakchats) [reply js/tiles.js]: In line 1010?

**Comment by niccokunzmann** [comment]
@saarthakchats Thanks, it looks good to me now. I merge. Please test after this if all is as expected.

**Comment by saarthakchats** [comment]
Ok thanks for merging

**Pull request merged**